File: Week_06/647/palindromic_substrings2.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        n = len(s)
        res = 0
        for i in range(2 * n - 1):
            l = i // 2
            r = l + i % 2
            while l >= 0 and r < n and s[l] == s[r]:
                res += 1
                l -= 1
                r += 1
        return res

File: Week_06/647/palindromic_substrings2.go
package main

func countSubstrings2(s string) int {
	n := len(s)
	res := 0
	for i := 0; i < 2*n-1; i++ {
		l := i /2
		r := l + i % 2
		for l >=0 && r <n && s[l] == s[r] {
			res++
			l--
			r++
		}
	}
	return res
}
